### ISSUE TITLE
Selecting an entry does nothing (Arch, WSL, Debian)

### ISSUE THREAD

**Issue by deathtrip** [dvorka/hstr#236]
I'm on Arch Linux x64 and when i select a history entry using hstr, nothing happens, the command doesn't execute.
It started happening while using 1.21, after an upgrade to 1.23 it still happens.

**Comment by rickywu** [comment]
The same problem on WSL:
hh version "1.23"
   build   "Sep 30 2017 11:20:10"

**Comment by hThoreau** [comment]
I've tried agoing as back as 1.19 and gost the same error, it seems that using TAB as a selector actually cancels it. I'm using bash verion 4.4.12 in a Debian Stretch

**Comment by dvorka** [comment]
I think that several different issues are mixed in this issue:

* **WSL**
    * Please make sure that you [configured](https://github.com/dvorka/hstr/blob/master/CONFIGURATION.md) HSTR - it's not needed on most Linux distributions, but on Windows HSTR will NOT work without it. 
* **Arch**
    * I need to install [VM](https://wiki.archlinux.org/index.php/VirtualBox) and try to reproduce the problem - can you please share 
* **Debian**
    * Can you please try <kbd>Enter</kbd> (completes and executes)? Can you also try right arrow (just completes)? I will try to reproduce the issue on Debian VM in the meantime.

**Comment by rickywu** [comment]
h --show-configuration >> ~/.bashrc
. .bashrc

Still not work

**Comment by hThoreau** [comment]
Hi @dvorka,
Sorry for taking so long to respond. I've jus tested the alternative selection options it suggested and it had just returned the following result:
```
user@subgraph:~$  hh -- 

user@subgraph:~$  hh -- 
```
Regards,

**Comment by pavelek** [comment]
Hi @dvorka,

The same problem as above ... ubuntu on Windows. Doesn't matter if run as "wsl hh" or "hh" from bash. Enter, TAB canceling command like left key
Right key passing empty command
Regards,

**Comment by balor** [comment]
+1

**Comment by dvorka** [comment]
Just want to let you know that I'm aware of the WSL (Ubuntu on Windows) bug. The problem is that I had hard times to make HSTR work under WSL - I had to use a different trick to get selected history item on command. The reason is that WSL terminal is incomplete - several key terminal features are simply not implemented under Windows - and Microsoft developers are fixing/changing the functionality. 

* check #186
* see also https://github.com/Microsoft/WSL/issues/2332

Also I'm not Windows user and to fix HSTR I had to find a Windows machine where I can fix the problem.

Please stay tuned.

**Comment by dvorka** [comment]
I just verified that HSTR works as expected on Debian. I was able to **reproduce** the problem on WSL - console is NOT fixed, however I found wrong command (hh instead of hstr) in .bashrc companion. HSTR works for me on WSL with this change. Fix will be released with the next version.

**Comment by iamjackg** [comment]
Hey, 

I have this all set up and it's working _almost_ fine on WSL. I compiled it from scratch, and I have symlinked `hh` to `hstr` so it gets called with the right name. The only remaining issue is that once I press Enter, the command appears on the command line but I have to press Enter _again_ to execute the command.

Is this expected? Is it just a limitation of the READLINE trick you're using to write to the terminal?
